rfd/0086/multiprocess.md

```diff
@@ -53,35 +58,44 @@ _Related GitHub issues:_
+ContainerPilot 3 will support the ability for a job to have multiple health checks. All health checks for a job must be in a passing state before a job can be marked as healthy. The job definition will be separated from the health check definition in order to support proposed [configuration improvements](config.md).
 
 We'll make the following changes:
 
-- ContainerPilot will maintain state for all services, which can be either `healthy` or `unhealthy`, and all health checks, which can be either `passing` or `failing`.
-- All health checks must be marked `passing` before ContainerPilot will mark the service a `healthy`.
-- A service in a `healthy` state will send heartbeats (a `Pass` message) to the discovery backend every `heartbeat` seconds with a TTL of `ttl`.
-- A health check will poll every `poll` seconds, with a timeout of `timeout`. If any health check fails (returns a non-zero exit code or times out), the associated service is marked `unhealthy` and a `Fail` message is sent to the discovery service.
-- Once any health check fails, _all_ health checks need to pass before the service will be marked healthy again. This is required to avoid service flapping.
-
-**Important note:** end users should not provide a health check with a long polling time to perform some supporting task like backends. This will cause "slow startup" as their service will not be marked healthy until the first polling window expires. Instead they should create another service for this task.
-
-```
-consul:
-  host: consul.svc.triton.zone
-
-services:
-  nginx:
-    port: 80
-    heartbeat: 5
-    ttl: 10
-
-health:
-  nginx:
-    check-A:
-      command: curl -s --fail localhost/health
-    check-B:
-      command: curl -s --fail localhost/otherhealth
-
+- ContainerPilot will maintain state for all jobs, which can be either `healthy` or `unhealthy`, and all health checks, which can be either `passing` or `failing`.
+- All health checks must be marked `passing` before ContainerPilot will mark the job as `healthy`.
+- A job in a `healthy` state will send heartbeats (a `Pass` message) to the discovery backend every `heartbeat` seconds with a TTL of `ttl`.
+- A health check will poll every `poll` seconds, with a timeout of `timeout`. If any health check fails (returns a non-zero exit code or times out), the associated job is marked `unhealthy` and a `Fail` message is sent to the discovery job.
+- Once any health check fails, _all_ health checks need to pass before the job will be marked healthy again. This is required to avoid service flapping.
+
+**Important note:** end users should not provide a health check with a long polling time to perform some supporting task like backends. This will cause "slow startup" as their job will not be marked healthy until the first polling window expires. Instead they should create another job for this task.
```

Comment:
To clarify my understanding: health checks should be short, and end users should never mix concerns between health checks and cron events. This is a good principle, but it's a also a practical limitation because a job will not be marked as healthy if the `job` runs long doing other work unrelated to the healthiness of the app. Yes?